openapi-core/src/main/java/io/ballerina/openapi/core/generators/common/OASModifier.java

```diff
@@ -268,19 +290,67 @@ private static String updateParameters(String pathValue, Map<String, String> par
                                            List<Parameter> parameters, List<Parameter> modifiedParameters) {
 
         for (Parameter parameter : parameters) {
-            if (!parameter.getIn().equals("path")) {
-                modifiedParameters.add(parameter);
-                continue;
+            String location = parameter.getIn();
+            if ("path".equals(location)) {
+                String modifiedPathParam = parameterNames.get(parameter.getName());
+                parameter.setName(modifiedPathParam);
+                pathValue = replaceContentInBraces(pathValue, modifiedPathParam);
+            } else if ("query".equals(location) || "header".equals(location)) {
```

Comment:
IMO we should have predefined constants for `path`, `query`, and `header` references.

Reply:
Fixed via [c0360b9](https://github.com/ballerina-platform/openapi-tools/pull/1770/commits/c0360b9e7086ebe02f649d1b10c3a2b34f392971)